pyproject.toml

```diff
@@ -84,5 +68,36 @@ version-header = "="
 version-strings = ["setup.py"]
 build-system = "setuptools"
```

Comment:
@justinmayer I think this need a change as the version will be only in this file and build-system will be changed too. Is this possible with `autopub`?

Reply:
I don't understand. Can you elaborate?

Reply:
Two things: 
I don't know autopub and if it is needed. If I should build a pipeline, my deploy step would be running `pdm build` and `pdm publish` based on branch/tag criteria. So my question would be what does autopub adds to that pipeline?
Second question:
version string is set to setup.py a file the will be removed in one of the next commits as all information is in pyproject.toml. The build system needs to be changed as well. Since you are the author of autopub, you can give me perhaps a hint on what to add there or if this should/ can be dropped.

Reply:
We use [AutoPub](https://justinmayer.com/projects/autopub/) to automatically publish releases. You can read more about it via that link, including why I created it. I believe it obviates the need for calling `pdm build` and `pdm publish`, neither of which are likely necessary in an AutoPub context.

You can see the existing `autopub build` step in the GitHub Actions workflow. That invocation runs [`python -m build`](https://github.com/autopub/autopub/blob/d89ef5a7a894fbfbcefe2d46e1e4dbc2b08fdc81/autopub/build_release.py#L14) ([docs for `build`](https://pypa-build.readthedocs.io/en/latest/)), a build front-end that uses the build backend specified in `pyproject.toml` to build the `sdist` and wheel. We use the PyPI Trusted Publisher system to actually publish the package to PyPI ([relevant configuration](https://github.com/getpelican/pelican/blob/fad2ff7ae3cd0ea8b974db5fe42de7383da679c1/.github/workflows/main.yml#L127-L129)), which is why `pdm publish` is not necessary.

Going forward the version number should only ever exist in `pyproject.toml` and will be automatically incremented by AutoPub based on the `Release type:` value specified in `RELEASE.md` files included in pull requests.

This might be easier in an interactive conversation. Can you come find me in the chat room? https://web.libera.chat/?#pelican